WcaOnRails/app/javascript/edit-schedule/SchedulesEditor/index.jsx

```diff
@@ -38,13 +39,21 @@ export class SchedulesEditor extends React.Component {
     };
   }
 
+  componentDidMount() {
+    // We cannot handle well changes (such as room color) when fullCalendar's element is hidden.
+    // So we unselect the room when our panel becomes hidden, to avoid running into any visual bug.
+    $(schedulesEditPanelSelector).find('.panel-collapse').on('hidden.bs.collapse', this.resetSelectedRoom);
```

Comment:
This is something I added after reviews.
I noticed that opening a room's schedule then change its color wouldn't update the currently shown calendar (ie: it keeps the old color).
If we keep the color in the props/state, and therefore update the fullcalendar when the color has changed, then we run into a pretty serious issue: fullcalendar's element is not visible at the time of the change, and calling a change of option on it makes it render with its current height: 0... (making it effectively invisible)

The workaround I found was to reset the selected room when the editor's panel is hidden.